src/Utilities/PrettyType.hpp

```diff
@@ -684,4 +684,21 @@ template <typename T>
 std::string get_runtime_type_name(const T& x) {
   return boost::core::demangle(typeid(x).name());
 }
+
+/*!
+ * \ingroup PrettyType
+ * \brief Extract the "base name" from a name, that is, the name
+ * without template parameters or scopes.
+ */
+std::string extract_base_name(std::string name);
```

Comment:
I cannot think of an alternative at the moment, but when I first saw base_name I thought about the name of a base class...